rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,211 @@
+
+One case where this difference may be functionally relevant is during disruptions when Glides is sending updates for trips that may not match the GTFS schedule data that RTR has. In these cases, including trip metadata like origin / destination and start / end times could provide RTR with additional context if we still want to attempt some simple trip matching on the RTR side.
+
+# Prior art
+[prior-art]: #prior-art
+
+The main prior art for this proposal comes from other MBTA modes, in particular bus and heavy rail.
+
+In the case of bus, operators sign in with a run number, thereby indicating the work that they will be doing. Stateful logic in TransitMaster and Swiftly is then able to combine the run schedule with realtime location tracking to update the assigned trip that a bus is operating. With some minor subtlety around a handful of individual trips that are broken up across multiple routes, all systems have the same trips and trip IDs from HASTUS. Major modifications to bus schedule data by `gtfs_creator` are rare, usually amounting at most to minor corrections to shapes or information about stops.
+
+At the other extreme, heavy rail tracking and trip assignment is mostly centralized in the OCS. Rather than having operator sign-ins on on-board hardware, dispatchers assign a vehicle to a trip and the OCS simply updates the trip assignments based on schedules as vehicles reach their terminals and complete trips. RTR still performs its own supplementary trip assignment logic on top of this in order to handle cases like vehicles deviating from the expected course of their scheduled trip.
+
+Of the two models, the proposed end state on light rail most closely resembles the current state on heavy rail. We currently lack the on-vehicle hardware to implement a bus-like solution for light rail, but Glides already exists and provides a means for other operations personnel to provide us with information about which trips are being operated by what vehicles.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+For the purposes of this RFC, we have had to assume that there will still be planned disruptions that have not been modeled in HASTUS in a way that reflects actual operations and trainsheets. This leaves open questions as to whether this will always be the case, and whether or not we will one day be able to assume that the schedule from HASTUS reflects planned operations in all cases.
```

Comment:
Yes, unplanned disruptions are an important consideration that I haven't really addressed explicitly in this RFC. My hope is that edits to trainsheets and things like dropped and added trips would flow through smoothly and RTR would at least be able to use that information effectively. That said, I did have a literal shower thought last night that, while I discuss trip to trip linking for scheduled trips that exist in HASTUS and future TODS, I don't think we really have a way to link added trips aside from via car numbers if we choose to go that route, so that's worth considering.